tooling-cpu-load/src/main/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/utils/LinuxProcessUtils.java

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * Priority of scheduling process on CPU, can be set in range -20 to 20, lower number = higher priority
+     */
+    public static final int HIGHEST_CPU_PROCESS_PRIORITY = -20;
+    public static final int NORMAL_CPU_PROCESS_PRIORITY = 0;
+    public static final int LOWEST_CPU_PROCESS_PRIORITY = 20;
+
+    public static final String LINUX_OS_NAME = "Linux";
+
+    /**
+     * Retrieves process id of EAP/Wildfly server using managament client
+     *
+     * @param client management client connected to server
+     * @return pid of the server
+     */
+    public static int getProcessId(OnlineManagementClient client) throws IOException {
+        ModelNode model = new ModelNode();
+        model.get(ClientConstants.OP).set("read-resource");
+        model.get(ClientConstants.OP_ADDR).add("core-service", "platform-mbean");
+        model.get(ClientConstants.OP_ADDR).add("type", "runtime");
+
+        ModelNodeResult result = client.execute(model);
+        String nodeName = result.get("result").get("name").asString();
+        return Integer.valueOf(nodeName.substring(0, nodeName.indexOf("@")));
+    }
+
+    /**
+     * Returns pid of given process.
+     *
+     * @param process process
+     * @return pid of the process
+     */
+    public static int getProcessId(Process process) {
+
+        int pid;
+        try {
+            Field f = process.getClass().getDeclaredField("pid");
+            f.setAccessible(true);
+            pid = f.getInt(process);
+        } catch (Throwable e) {
+            throw new IllegalStateException("Unable to get process id for " + process.getClass().getSimpleName(), e);
+        }
+
+        if (pid == 0) {
+            throw new IllegalStateException("Process ID is 0 for " + process.getClass().getSimpleName());
+        }
+
+        return pid;
+    }
+
+    /**
+     * Priority can be set -19 to 20, lower number = higher priority
+     *
+     * @param pid id of the process
+     * @param priority priority to be set, -19 to 20, lower number = higher priority
+     */
+    public static void setPriorityToProcess(int pid, int priority) throws Exception {
+
+        if (isPriorityInRange(priority)) {
+            if (!isLinuxEnvironment()) {
+                throw new UnsupportedOperationException(
+                        "Command renice which is used to lower priority of process is supported only on linux. " +
+                                "Current operation system is: " + System.getProperty("os.name"));
+            }
+
+            Process setPriorityProcess = Runtime.getRuntime().exec("renice -n " + priority + " -p " + pid);
+            if (setPriorityProcess.waitFor() != 0) {
+                throw new Exception("Process: " + setPriorityProcess + " did not exit with code 0.");
+            }
+        } else {
+            throw new Exception(
+                    "Set process priority between " + LOWEST_CPU_PROCESS_PRIORITY + " to " + HIGHEST_CPU_PROCESS_PRIORITY
```

Comment:
IMHO this exception contains only a message which is a suggestion for the user but not the actual error.